### ISSUE TITLE
`Error: Unexpected HTTP response: 404` error

### ISSUE THREAD

**Issue by alexcb** [earthly/actions-setup#23]
I got this error when using 1.0.5 version

```Matched version: v1.0.5
earthly cache is not found
Successfully deleted pre-existing /home/runner/.earthly/bin
Error: Unexpected HTTP response: 404
```

_Originally posted by @drselump14 in https://github.com/earthly/actions-setup/issues/22#issuecomment-1501361596_

**Comment by alexcb** [comment]
This error occurs when pinning to earthly `0.7.1`.

**Comment by alexcb** [comment]
@drselump14 if you try using the latest (v1.0.7) of actions-setup, it'll now display the URL (of earthly) that it is attempting to download. It would be helpful to know what URL it's getting a 404 on.

**Comment by alexcb** [comment]
We sometimes get this error when a release is underway; the script detects a new release version exists; however not all binaries have completed being uploaded.

we should add some sort of sleep/retry loop for this case.

**Comment by idodod** [comment]
This should be now fixed, will be included in next release - v1.0.13
